File: mvp/src/main/java/cn/cuibg/mvp/MvpPresenter.java
package cn.cuibg.mvp;

import android.content.Intent;

import androidx.annotation.UiThread;

public interface MvpPresenter<V extends MvpView> {

    @UiThread
    void attachView(V view);

    @UiThread
    void detachView(boolean retainInstance);

    @UiThread
    void create();

    @UiThread
    void resume();

    @UiThread
    void pause();

    @UiThread
    void stop();

    @UiThread
    void destroy();

    @UiThread
    void activityResult(int requestCode, int resultCode, Intent data);
}

File: mvp/src/main/java/cn/cuibg/mvp/MvpPresenter.kt
package cn.cuibg.mvp

import android.content.Intent
import androidx.annotation.UiThread

interface MvpPresenter<V : MvpView?> {
    @UiThread
    fun attachView(view: V)

    @UiThread
    fun detachView(retainInstance: Boolean)

    @UiThread
    fun create()

    @UiThread
    fun resume()

    @UiThread
    fun pause()

    @UiThread
    fun stop()

    @UiThread
    fun destroy()

    @UiThread
    fun activityResult(requestCode: Int, resultCode: Int, data: Intent?)
}